### PR TITLE
dm: Improve API documentation

### PR DESCRIPTION
**Pull request by maje-emb**
Added improvements to the Distance Measurement submodule API documentation. Improved doxygen markup.

### CODE REVIEW

File: include/dm.h [hunk 1]
```diff
@@ -17,79 +23,78 @@
...
-	/* Poor measurement quality */
+	/** Poor measurement quality. */
 	DM_QUALITY_POOR,
 
-	/* Measurement not for use */
+	/** Measurement not for use. */
 	DM_QUALITY_DO_NOT_USE,
 
-	/* Incorrect CRC measurement */
+	/** Incorrect CRC measurement. */
```
Comment (grochu) [inline include/dm.h:58]: How about:
```suggestion
	/** Incorrect measurement CRC. */
```

**Review by grochu — COMMENTED** [review]
Thanks for the update, it looks good. Just added one suggestion for improvement.

File: include/dm.h (hunk 1 above)
Comment (maje-emb) [reply include/dm.h]: Good point.

File: include/dm.h [hunk 2]
```diff
@@ -17,79 +23,78 @@
...
-#include <zephyr/bluetooth/addr.h>
-
 /** @brief Role definition. */
 enum dm_dev_role {
-	/* Undefined role*/
+	/** Undefined role. */
 	DM_ROLE_NONE,
 
-	/* Act as a initiator */
+	/** Act as a initiator. */
```
Comment (peknis) [inline include/dm.h:31]:
```suggestion
	/** Act as an initiator. */
```

File: include/dm.h [hunk 3]
```diff
@@ -17,79 +23,78 @@
...
-	/* Good measurement quality */
+	/** Good measurement quality. */
 	DM_QUALITY_OK,
 
-	/* Poor measurement quality */
+	/** Poor measurement quality. */
 	DM_QUALITY_POOR,
 
-	/* Measurement not for use */
+	/** Measurement not for use. */
```
Comment (peknis) [inline include/dm.h:55]:
```suggestion
	/** Measurement not in use. */
```
Not so sure about this, but maybe it tries to say the measurement is not available?

File: include/dm.h [hunk 4]
```diff
@@ -17,79 +23,78 @@
...
-	/* Measurement quality not specified */
+	/** Measurement quality not specified. */
 	DM_QUALITY_NONE,
 };
 
-/** @brief Measurement structure. */
+/** @brief Distance Measurement result structure. */
 struct dm_result {
-	/* Status of the precedure. \a true if measurement was done correctly, \a false otherwise */
+	/** Status of the procedure. \a true if measurement was done correctly,
```
Comment (peknis) [inline include/dm.h:67]:
```suggestion
	/** Status of the procedure. \a True if measurement was done correctly,
```

File: include/dm.h [hunk 5]
```diff
@@ -17,79 +23,78 @@
...
+	 * \a false otherwise.
+	 */
 	bool status;
 
-	/* Quality indicator */
+	/** Quality indicator */
 	enum dm_quality quality;
 
-	/* Bluetooth LE Device Address */
+	/** Bluetooth LE Device Address. The address is used to distinguish peers. */
```
Comment (peknis) [inline include/dm.h:75]: Does Device Address need to be with capital initial letters?

File: include/dm.h [hunk 6]
```diff
@@ -143,10 +152,10 @@ int dm_init(struct dm_init_param *init_param);
  *
  *  Adding a measurement request. This is related to timeslot allocation.
  *
- *  @param req Address of the structure with request parameters.
+ *  @param[in] req Address of the structure with request parameters.
  *
  *  @retval 0 If the operation was successful.
```
Comment (peknis) [inline include/dm.h:157]:
```suggestion
 *  @retval 0 if the operation was successful.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by grochu — APPROVED** [review]

File: include/dm.h (hunk 3 above)
Comment (maje-emb) [reply include/dm.h:55]: The idea is that the measurement is available (the callback function was called), but the measurement is poor quality, so it should not be used.

File: include/dm.h (hunk 5 above)
Comment (maje-emb) [reply include/dm.h]: I think that Device Address does not need to be with capital initial letters.
I also corrected it in other places.

File: include/dm.h (hunk 3 above)
Comment (peknis) [reply include/dm.h:55]: Then it should say so. For example, Bad measurement quality.

Comment (maje-emb) [reply include/dm.h]: Maybe: "Measurements not usable". Because we have a `DM_QUALITY_POOR` field in the enum structure and to avoid confusion.

**Review by peknis — APPROVED** [review]

File: include/dm.h [hunk 7]
```diff
@@ -7,7 +7,13 @@
 #ifndef DM_H_
 #define DM_H_
 
-/**@file
+#include <zephyr/kernel.h>
+#include <stdlib.h>
```
Comment (KAGA164) [inline include/dm.h:11]: Please move it above `<zephyr/kernel.h>` and add a blank line after. It will look more readable and nice

Comment (maje-emb) [reply include/dm.h:10]: ACK

**Review by KAGA164 — APPROVED** [review]

**Pull request merged**